### PR TITLE
add cookie path support to config via label istio.io/persistent-session

### PR DESCRIPTION
**Pull request by sanjaypujare**
**Please provide a description of this PR:**
This is an enhancement over the previous PR #40660 to add cookie path support via the label configuration

### CODE REVIEW

**Comment by sanjaypujare** [comment]
@costinm as discussed. PTAL

File: pilot/pkg/networking/grpcgen/lds.go [hunk 1]
```diff
@@ -298,6 +298,11 @@ func buildOutboundListeners(node *model.Proxy, push *model.PushContext, filter l
 				if features.PersistentSessionLabel != "" {
 					sessionCookie := sv.Attributes.Labels[features.PersistentSessionLabel]
 					if sessionCookie != "" {
+						cookieNamePath := strings.Split(sessionCookie, ":")
```
Comment (howardjohn) [inline pilot/pkg/networking/grpcgen/lds.go:301]: nit: `strings.Cut` is preferred for splitting at a single element, although the behavior should be the same in this case

**Review by howardjohn — COMMENTED** [review]
seems reasonable to be, deferring to @costinm though

**Review by costinm — APPROVED** [review]

**Comment by sanjaypujare** [comment]
@costinm or @howardjohn could one of you trigger the tests so this PR can be merged?

**Comment by sanjaypujare** [comment]
Thank you @howardjohn and @costinm !

**Pull request merged**